src/main/java/com/softserveinc/ita/pageobjects/BasePage.java

```diff
@@ -93,4 +93,18 @@ public T closeAdvertisingBannerIfDisplayed() {
 
         return (T) this;
     }
+
+    public ComparisonPage openComparisonPage() {
+        $("rz-comparison button")
+                .shouldBe(visible)
+                .hover()
+                .click();
+        $("a.comparison-modal__link").click();
+
+        return new ComparisonPage();
+    }
+
+    public String getUrlText() {
```

Comment:
create util class and name it something like BrowserTabHelper and move there this method and rename it to getCurrentUrl()

and in place, you will need this method to call it through a variable or just new BrowserTabHelper().getCurrentUrl();

Reply:
maybe it's better to do it like: `var currentUrl = WebDriverRunner.url()`, with static import it's gonna look like: 
`var currentUrl = url()`. This way i don't need to create util class for this method

Reply:
for operations such as moving between windows-tabs and getting URL or refresh better to have a separate class for this and/or perform them in tests instead of page objects

you moved it into the test and it's ok, static import is fine

Reply:
got it